common/core/desktop/build.sh

```diff
@@ -234,65 +262,90 @@ build_windows() {
   fi
 }
 
-build_linux_macos() {
+build_standard() {
+  local BUILD_PLATFORM="$1"
+  local ARCH="$2"
+  local RUSTARCH=${3:-}
+  if [ $# -gt 3 ]; then
+    shift 3
+    local STANDARD_MESON_ARGS="$*"
+  else
+    local STANDARD_MESON_ARGS=
+  fi
 
   # Build rust targets
-  build_test_rust arch
+  build_test_rust "$ARCH" "$RUSTARCH"
 
   # Build meson targets
   if $CONFIGURE; then
-    echo_heading "======= Configuring C++ library for $os_id ======="
-    pushd $THIS_DIR > /dev/null
-    meson $MESON_PATH --werror --buildtype $MESON_TARGET $ADDITIONAL_ARGS
+    echo_heading "======= Configuring C++ library for $BUILD_PLATFORM ======="
+    pushd "$THIS_DIR" > /dev/null
+    meson setup "$MESON_PATH" --werror --buildtype $MESON_TARGET $STANDARD_MESON_ARGS $ADDITIONAL_ARGS
     popd > /dev/null
   fi
 
   if $BUILD_CPP; then
-    echo_heading "======= Building C++ library for $os_id ======="
-    pushd $MESON_PATH > /dev/null
+    echo_heading "======= Building C++ library for $BUILD_PLATFORM ======="
+    pushd "$MESON_PATH" > /dev/null
     ninja
     popd > /dev/null
   fi
 
   if $TESTS_CPP; then
-    echo_heading "======= Testing C++ library for $os_id ======="
-    pushd $MESON_PATH > /dev/null
+    echo_heading "======= Testing C++ library for $BUILD_PLATFORM ======="
+    pushd "$MESON_PATH" > /dev/null
     meson test --print-errorlogs
     popd > /dev/null
   fi
 
   if $INSTALL_RUST; then
-    echo_heading "======= Installing Rust libraries for $os_id ======="
+    echo_heading "======= Installing Rust libraries for $BUILD_PLATFORM ======="
     # TODO
   fi
 
   if $INSTALL_CPP; then
-    echo_heading "======= Installing C++ libraries for $os_id ======="
-    pushd $MESON_PATH > /dev/null
-    ninja install
+    echo_heading "======= Installing C++ libraries for $BUILD_PLATFORM ======="
+    pushd "$MESON_PATH" > /dev/null
+    meson install
```

Comment:
What's the reason for calling `meson` instead of `ninja` here? The meson version available on Bionic (0.45.1) is too old for this to work:

```
eberhard@bionic:~/Projects/keyman/common/core/desktop/build/arch/release$ DESTDIR=/tmp/foo meson install
Error during basic setup:

Neither directory contains a build file meson.build.
```

I see the following workarounds:

a) revert to use `ninja` again (but this might not be possible on other platforms)
b) check the meson version. If it's < 0.53.2 use ninja, otherwise meson (0.53.2 is what we have on Focal)

Of course, trying to package a newer meson version for Bionic would also be an option but with a lot more work.

Reply:
I changed to `meson install` for consistency. `ninja install` should still work... argh. inconsistent versions of build tools is monstrous!